daliuge-engine/dlg/apps/mpi.py

```diff
@@ -66,33 +76,105 @@ class MPIApp(BarrierAppDROP):
     This drop will gather the individual exit codes from the launched
     applications and transition to ERROR if any of them did not exit cleanly,
     or to FINISHED if all of them finished successfully.
+
     """
 
+    input_parser: DropParser = dlg_enum_param(DropParser, "input_parser", DropParser.PICKLE)  # type: ignore
+
     def initialize(self, **kwargs):
         super(MPIApp, self).initialize(**kwargs)
 
-        self._command = self._popArg(kwargs, "command", None)
         self._maxprocs = self._popArg(kwargs, "maxprocs", 1)
         self._use_wrapper = self._popArg(kwargs, "use_wrapper", False)
-        self._args = self._popArg(kwargs, "args", [])
+        self._args = self._popArg(kwargs, "args", "")
+        self._applicationArgs = self._popArg(kwargs, "applicationArgs", {})
+        self._argumentPrefix = self._popArg(kwargs, "argumentPrefix", "--")
+        self._paramValueSeparator = self._popArg(kwargs, "paramValueSeparator", " ")
+        self._inputRedirect = self._popArg(kwargs, "input_redirection", "")
+        self._outputRedirect = self._popArg(kwargs, "output_redirection", "")
+
+        self._command = self._popArg(kwargs, "command", None)
         if not self._command:
             raise InvalidDropException(
                 self, "No command specified, cannot create MPIApp"
             )
+        self._recompute_data = {}
 
     def run(self):
         from mpi4py import MPI
 
         cmd, args = self._command, self._args
+        inputs = self._inputs
```

Comment:
Is there a reason why we use this indirect approach, rather than just access `self._inputs` directly? It hides the fact that we are updating the member variables for no clear benefit.